src/hooks/useDecodedId.tsx

```diff
@@ -0,0 +1,10 @@
+import { useSearchParams } from 'react-router-dom'
+
+const useDecodedId = () => {
+  const [searchParams] = useSearchParams()
+  const encodedId = searchParams.get('id')
+  const decodedId = encodedId ? atob(encodedId) : null
+  return decodedId
+}
```

Comment:
I think we should return at the line number 5, avoid to create variable which will using one time